smart/model.py

```diff
@@ -998,6 +1005,71 @@ def _init_4_7_set_initial_conditions(self):
                     u_cur.vector().set_local(values)
                     u_cur.vector().apply("insert")
 
+        for parameter in self.pc:
+            if parameter.type == ParameterType.from_xdmf:
+                # load the time vec from xdmf file
+                xdmf_file = open(parameter.xdmf_file, "r")
+                xdmf_string = xdmf_file.read()
+                found_pattern = re.findall(r"Time Value=\"?[^\s]+", xdmf_string)
+                tVec = []
+                for i in range(len(found_pattern)):
+                    tVec.append(float(found_pattern[i][12:-1]))
+                parameter.tVec = np.array(tVec)
+
+                # define function space
+                if parameter.compartment not in self.cc.keys:
+                    raise ValueError(
+                        f"Compartment name {parameter.compartment} for parameter"
+                        f"{parameter.name} does not match a known compartment"
+                    )
+                V_cur = d.FunctionSpace(self.cc[parameter.compartment].dolfin_mesh, "P", 1)
+                parameter.dolfin_function = d.Function(V_cur)
+
+                cur_file = d.HDF5File(self.parent_mesh.mpi_comm, parameter.h5_file, "r")
+                # Find index associated with the starting time
```

Comment:
Doing this and conditionally might be a bit dangerous because you might forget to close the file which might lead to strange bugs